nbconvert/nbconvertapp.py

```diff
@@ -323,16 +339,20 @@ def init_single_notebook_resources(self, notebook_filename):
 
         return resources
 
-    def export_single_notebook(self, notebook_filename, resources):
+    def export_single_notebook(self, shell_input, resources):
         """Step 2: Export the notebook
 
         Exports the notebook to a particular format according to the specified
         exporter. This function returns the output and (possibly modified)
         resources from the exporter.
 
+        shell_input: a filename or buffer from stdin depending on the '--stdin' option
```

Comment:
the butter does not need to be from stdin, so you can say "for example from stdin when the the '--stdin' is used", 
but as a API, you do not **need** to be stdin.